lib/lrama/diagram.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "erb"
+begin
+  require "railroad_diagrams"
+rescue LoadError
+  warn "railroad_diagrams is not installed. Please run `bundle install`."
+end
```

Comment:
If you're using it in `ruby/ruby`, the library may not be installed by default. In such cases, you need to handle it like this:  

This ensures that the script doesn't break if the gem isn't available, while also notifying the user about the missing dependency.

But if there is a better way, I would like to know.

Reply:
Yes, we should not enforce any installation of gems for ruby/ruby build.
However current implementation causes warnings on ruby/ruby build (see:  https://github.com/ruby/lrama/actions/runs/13088787136/job/36523059170?pr=523#step:14:76), we also need to avoid such situation.
When I implemented [stackprof integration](https://github.com/ruby/lrama/blob/88097af10d937c5c6a2890a9be75a5b27f0d1c8d/lib/lrama/report/profile.rb#L8), I postponed `require` until entry point method is called.

Then my suggestions are:
1. extract `require` logic (L. 4 - 8) to an instance method
2. call the instance method when it's necessary, it might be needed when `Diagram#initialize` is called

By the way, we can assume "erb" is bundled to ruby then please keep `require "erb"` as is.

Reply:
https://github.com/ruby/lrama/blob/1df1c1b5b3a4ea265415d134c9bfb7991887ae39/lib/lrama/report/profile/call_stack.rb#L24 is an example.

Reply:
Thank you! I updated https://github.com/ruby/lrama/pull/523/commits/9113e0b873664f7eb1306ded809474e418251967